npm/podController.go

```diff
@@ -106,7 +93,23 @@ func (nPod *NpmPod) removeLabelsWithKey(key string) {
 }
 
 func (nPod *NpmPod) appendContainerPorts(podObj *corev1.Pod) {
-	nPod.ContainerPorts = getContainerPortList(podObj)
+	portList := getContainerPortList(podObj)
+	// To remove any references to objects sent to events by sharedinformer,
+	// NPM will need to deepcopy data from the provided objects
+	copiedPortList := make([]corev1.ContainerPort, len(portList))
```

Comment:
getContainerPortList returns new slices with copy. Why do we need the same operation again?